src/stormpath.social-login.js

```diff
@@ -0,0 +1,187 @@
+(function() {
+  'use strict';
+
+  /**
+   * @ngdoc overview
+   *
+   * @name  stormpath.socialLoginService
+   *
+   * @description
+   *
+   * This module provides the {@link stormpath.socialLoginService.$socialLogin $socialLogin} service.
+   *
+   * Currently, this provider does not have any configuration methods.
+   */
+  function SocialLoginService(STORMPATH_CONFIG, $injector, $http, $q) {
+    var providersPromise;
+
+    function initProviders(providers) {
+      Object.keys(providers).forEach(function(providerName) {
+        var provider = providers[providerName];
+        var service;
+
+        try {
+          service = $injector.get('$' + providerName + 'Login');
+        } catch (err) {
+          return;
+        }
+
+        service.clientId = provider.clientId;
+        providers[providerName].service = service;
+      });
+    }
+
+    return {
```

Comment:
Same as mentioned before. Bad practice to have a constructor return another object than itself. Since `new SocialLoginService() != instanceof SocialLoginService`.